src/components/StateMachine.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import React, { Component } from 'react';
+
+export type ChangeStateFunc = (nextState: string, data?: Object) => void;
+
+const createStateMachine = (initialState, states) => {
+  return () => {
+    class StateMachine extends Component {
```

Comment:
Creo que en esta clase podemos implementar el listener de firebase `ref.on('value', this.onChange)` y este actualizaría el estado interno de react.

Reply:
El listener se inicia en `componentDidMount()`